docs/man/rpm-dependency-generators.7.scd

```diff
@@ -0,0 +1,381 @@
+only kept in memory during package builds. A single file may have an arbitrary
+number of attributes, and a single attribute may be assigned to an arbitrary
+number of files.
+
+A valid file attribute called _NAME_ is introduced with an *rpm-macrofile*(5),
+_NAME.attr_, dropped into the *%{\_fileattrsdir}* directory, that defines at
+least one of the following macros:
+
+```
+%__NAME_path
+%__NAME_magic
+%__NAME_exclude_path
+%__NAME_exclude_magic
+```
+
+The values of these macros represent the inclusion and exclusion rules. They are
+macro-expanded on use and then interpreted as extended regular expressions (RE).
+Matching is inclusive unless changed by _flags_ (see below): If the path, magic
+or both match, the file is said to _have_ the attribute in question, unless
+there is a matching exclusion rule (also inclusive by default) or a flag that
+prevents the match.
+
+The path REs are matched against the packaged path, that is, without the
+*%buildroot* prefix (for example, _/bin/bash_). The magic REs are matched
+against the result of *magic*(5) classification and possibly the *FILECLASS* tag
+stored in the package (see *--fileclass* in *rpm*(8) for details).
+
+Flags are comma-separated lists specified with the following macros:
+
+```
+%__NAME_flags
+%__NAME_exclude_flags
+```
+
+Currently, the following flags are supported (Added: 4.9.1):
+- *exeonly*: require executable bit set
+- *magic_and_path*: require both magic and path to match
+
+# GENERATORS
+
+A generator is an executable that reads filename(s) from stdin and writes
+dependencies of a single _type_ to stdout, one per line. Unlike the path REs,
+generators receive filenames with the *%buildroot* prefix since they typically
+need to look at the actual file contents on disk. This simple protocol allows
+for the generators to be implemented in any preferred language and enables them
+to use language specific libraries or tools.
+
+Generators are written _for_ file attributes. Thus, a valid generator for a file
+attribute _NAME_ is introduced with any of the following macros (one for every
+supported type) defined in the respective _NAME.attr_ file:
+
+```
+%__NAME_conflicts
+%__NAME_enhances
+%__NAME_obsoletes
+%__NAME_orderwithrequires
+%__NAME_provides
+%__NAME_requires
+%__NAME_recommends
+%__NAME_suggests
+%__NAME_supplements
+```
+
+The value of each macro is a command line (the program name and any arguments)
+that should be executed for every file having the _NAME_ attribute.
+
+Macros can also be passed to generators through the command line. For example,
+to pass the package name and version to the requires generator *foo*, define the
+macro:
+
+```
+%__foo_requires 	/some/script %{NAME} %{VERSION}
+```
+
+In addition to the globally defined macros, the following macros are exported to
+generators on a sub-package basis (Added: 4.15.0):
+
+```
+%{NAME}
+%{EPOCH}
+%{VERSION}
+%{RELEASE}
+```
+
+Note that the *OrderWithRequires* tag will only appear in RPM >= 4.17.0.
```

Comment:
This little cowboy is a long long way from home :smile:

The remark belongs to a section where different generators types are introduced, and should instead be in the form of "Added: 4.17.0" instead of this strange future expression.

And, that makes me think that we should have an explicit list of the supported generator (dependency) types. It's currently a bit hard to decipher. That's perhaps another point for using _NAME_DEPTYPE_foo in the SYNOPSIS.